racket-src/decl/decl-to-clause/trait-item.rkt

```diff
@@ -14,9 +14,82 @@
   ;; * The invariants that hold
   trait-item-decl-rules : CrateDecls CrateId (TraitId KindedVarIds_trait) TraitItem -> (Clauses Invariants)
 
-  [;; For a method declared in a trait: currently no logical rules are created.
-   (trait-item-decl-rules CrateDecls CrateId (TraitId KindedVarIds_trait) FnDecl)
-   ([] [])
+  [;; For a method declared in the trait Bar with parameter Self, like the following:
+   ;;
+   ;;     fn foo<T>(x: T) -> T where T: Ord { ... }
+   ;;
+   ;; We generate the following clauses
+   ;;
+   ;;     (∀ [(type Self) (type T)]
+   ;;         (well-formed-fn foo[Self T]) :-
+   ;;            (well-formed (type Self))
+   ;;            (well-formed (type T))
```

Comment:
I saw that associated types do not require well-formedness of their parameters. Why is that? Is it correct to require it for associated functions?